metadata-ingestion/docs/dev_guides/add_stateful_ingestion_to_source.md

```diff
@@ -0,0 +1,289 @@
+# Adding Stateful Ingestion to a Source
```

Comment:
this doc looks pretty well-written

one question (which may not be important enough to address in the doc) - what if I write a source that uses both stale entity removal and redundant run elimination?

Reply:
@hsheth2, The design allows for a source to use multiple use-cases via composition. The use-case handlers are independent of each other since they use different jobIds. The snowflake_v2 connector is an example of a source that uses both. See https://github.com/datahub-project/datahub/blob/255812939128de40af0b4edc033bad6877956a64/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py#L184.